### ISSUE TITLE
Spells cast from Magic Item tab of an Inventory item

### ISSUE THREAD

**Issue by CarnivorousM** [espositos/fvtt-tokenactionhud#54]
Windows 10, Foundry 0.6.5, DnD5e 0.94, Token Action HUD 0.5.9 (Using Minor QoL.)

When click into HUD (target(s) chosen and current character selected):
If I use a weapon for attacking - things go as planned (attack/damage applied/etc).
If I cast a spell from Spellbook - things go as planned (Templates/damage applied/etc).

If the item has charges and spells attached on Magic Item tab and I cast the spell from the "Magic Items" pulldown of the HUD - the spell is cast as a card (as if not using QoL). The charge seems to be tabulated, but the spell does not go off through QoL the same way it does as if it were cast as a regular spell from the Spellbook tab.

Example: Trying to use a magical wand that has 3 charges per day and can cast Cure Wounds using 1 charge. The spell for the Cure Wounds shows up in the HUD. If I click it, the spell 'fires' and I get a [Cure Wounds] card pop up with a second box allowing me to click the [damage] (healing) roll for it. This is not automated, nor does it apply the healing (even if I do click it) as it would be if I had Cure Wounds in my Spellbook and cast it from there (which DOES work). It does show that a change has been spent, though.

Not sure if I have overlooked something, or if there is a slight disconnect when casting from an item as opposed to casting from 'just a spell'.

Attached a screenshot of casting a healing spell from the Spells tab on HUD vs casting a healing spell from under Magic Items.

![Screen](https://user-images.githubusercontent.com/11615728/88617868-40727400-d05d-11ea-8351-f1c0af9149a8.jpg)

**Comment by espositos** [comment]
It does sound like a bug, I'll look into it. Thanks for the detailed report!

**Comment by CarnivorousM** [comment]
The card comes from dnd5e.
The action is not being processed through QoL the same way that a normal spell is processed.
If a spell (under Token HUD 'Spells' tab) is activated - QoL processes it fine.
When a Items spell is triggered (under Token HUD 'Magic Items' tab) - Qol does not seem to process these.

I assume there is something that 'points' either spell to be processed (since 1 of the settings for the module is to select 'Minor QoL Improvements' as a roll handler) and that 1 is doing it and the other is not.

Should the spells cast from magic items have the spells processed the same way that a Spellbook spell is processed?

**Comment by espositos** [comment]
It's more that I can't get Minor QoL to produce that chat card for me, even from the spellbook in the character sheet, so I don't know if that card is produced by Minor QoL or something else. Could you do me a favour create a test world with just Minor QoL and Magic Items enabled, set up a magic item with spells, and see if Minor QoL produces that card? If it does, could you send me a screenshot of your Minor QoL settings so I can try to replicate it.

**Comment by CarnivorousM** [comment]
This is interesting.
Setup a fresh (Blank) world. Only Token HUD, Magic Item and QoL active.
'Dude' was made a lvl 1 Wizard to allow spell slots and handed Cure Wounds. Hurt Dude was set to 2 hp for targeting.
Wand was created to cast Cure Wounds with 3 charges.

The first cast was from under HUD/Spells (using a spell slot) and is the first result (with QoL output in second box happening automatically).

Then cast from HUD/Magic Items (screen shot shows a charge was used (2/3) and I got the 3rd box.
Here's where it's different -I have to click on the 'Healing' roll of the 3rd box to trigger the actual roll. Once rolled - QoL did seem to pick it up. It just wasn't automated straight from the Item cast the way using a spell slot happened.
Before - I would get the same box as box 4 in the Blank World and then the default box to roll. This time, I only got the 1 box and once I click/rolled - QoL grabbed it.
![image](https://user-images.githubusercontent.com/11615728/88853267-13d17000-d1b5-11ea-91f8-e37066055e22.png)
